imap/xapian_wrap.cpp

```diff
@@ -181,22 +181,10 @@ static void remove_legacy_metadata(Xapian::WritableDatabase& db)
         // Remove legacy keys and values.
         if ((*key).find('.') != std::string::npos ||
             (!val.empty() && !isalpha(val[0]))) {
-            db.set_metadata(*key, "");
-        }
-    }
-    for (Xapian::docid docid = 1; docid <= db.get_lastdocid(); ++docid) {
-        try {
-            Xapian::Document doc = db.get_document(docid);
-            const std::string& val = doc.get_value(SLOT_DOCLANGS);
-            // Remove legacy doclang slot values.
-            if (!val.empty() && !isalpha(val[0])) {
-                doc.remove_value(SLOT_DOCLANGS);
-            }
-        }
-        catch (Xapian::DocNotFoundError e) {
-            // ignore
+            metadb.set_metadata(*key, "OVERWRITE:");
         }
     }
+    // NOTE: we're not removing DOCLANG slot because we can't tell the compactor to remove it
```

Comment:
Yeah, that's too bad indeed. We could do that in a two-pass compact if we encounter a db with a legacy language count slot. Our reindexed dbs shouldn't have one anyway.

Reply:
I'm not too stressed about that then, because we'll be reindexing everything up to v15 over the next few weeks.